### PR TITLE
Add Vector3, AbsoluteContentSize, and ScrollingFrame/misc properties

### PR DESCRIPTION
**Pull request by ckdavid**

### CODE REVIEW

File: lib/types/Vector3_spec.lua [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	end)
+
+	it("should have a constructor with three parameters", function()
+		local v = Vector3.new(1, 2, 3)
+
+		assert.not_nil(v)
+		assert.are.same({1, 2, 3}, extractValues(v))
+	end)
+
+	it("should throw when one or two parameters are passed", function()
```
[the remark this answers is not in the document]
Comment (ckdavid) [reply lib/types/Vector3_spec.lua]: I was following the model of what's in Vector2. Happy to remove this test or change it to reflect the behaviour. Is the actual behaviour that for each argument it defaults to 0 if it's not a number?

File: lib/instances/UIGridStyleLayout.lua [hunk 2]
```diff
@@ -3,10 +3,40 @@ local FillDirection = import("../Enum/FillDirection")
 local HorizontalAlignment = import("../Enum/HorizontalAlignment")
 local InstanceProperty = import("../InstanceProperty")
 local SortOrder = import("../Enum/SortOrder")
+local Vector2 = import("../types/Vector2")
 local VerticalAlignment = import("../Enum/VerticalAlignment")
 
 local UIGridStyleLayout = BaseInstance:extend("UIGridStyleLayout")
 
+UIGridStyleLayout.properties.AbsoluteContentSize = InstanceProperty.readOnly({
```
[the remark this answers is not in the document]
Comment (ckdavid) [reply lib/instances/UIGridStyleLayout.lua]: Yeah, it doesn't appear to be. This was just inspired by the checks done in GuiObject for AbsolutePosition and AbsoluteSize.

File: lib/types/Vector3_spec.lua (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kampfkarren) [reply lib/types/Vector3_spec.lua]: Yes, but I'd wait for Lucien to comment on this: he's fine with altering Lemur to be more strict in certain cases.

File: lib/instances/UIGridStyleLayout.lua (hunk 2 above)
[the remark this answers is not in the document]
Comment (Kampfkarren) [reply lib/instances/UIGridStyleLayout.lua]: I didn't notice it used the AbsoluteSize of the GuiObjects, which is where the check for that should be, whoops.

**Comment by LPGhatguy** [comment]
This is a fair bit to review all at once, sorry I haven't gotten to it as of two weeks!

For future PRs, can you break down changes into smaller PRs? Most of Lemur is fairly independent, so they shouldn't conflict normally.

**Comment by LPGhatguy** [comment]
That would actually be great, yeah! Then we don't have to block all of the features based on time or problems identified in individual pieces.

**Pull request closed without merging**